src/modules/Elsa.Workflows.Core/Services/ActivityRegistryLookupService.cs

```diff
@@ -0,0 +1,44 @@
+using Elsa.Workflows.Contracts;
+using Elsa.Workflows.Models;
+
+namespace Elsa.Workflows.Services;
+
+/// <summary>
+/// Represents a service used to lookup activity descriptors in the activity registry.
+/// </summary>
+public class ActivityRegistryLookupService(IActivityRegistry activityRegistry, IEnumerable<IActivityProvider> providers) : IActivityRegistryLookupService
+{
+    /// <inheritdoc />
+    public Task<ActivityDescriptor?> Find(string type)
+    {
+        return Find(() => activityRegistry.Find(type));
+    }
+
+    /// <inheritdoc />
+    public Task<ActivityDescriptor?> Find(string type, int version)
+    {
+        return Find(() => activityRegistry.Find(type, version));
+    }
+
+    /// <inheritdoc />
+    public Task<ActivityDescriptor?> Find(Func<ActivityDescriptor, bool> predicate)
+    {
+        return Find(() => activityRegistry.Find(predicate));
+    }
+
+    /// <inheritdoc />
+    public IEnumerable<ActivityDescriptor> FindMany(Func<ActivityDescriptor, bool> predicate)
+    {
+        return activityRegistry.FindMany(predicate);
+    }
+    
+    private async Task<ActivityDescriptor?> Find(Func<ActivityDescriptor?> findPredicate)
```

Comment:
Please mind the naming convention to apply the `Async` suffix when changing methods into async.